cpp/src/utilities/stream_pool.cpp

```diff
@@ -130,7 +130,6 @@ rmm::cuda_device_id get_current_cuda_device()
  */
 struct cuda_event {
   cuda_event() { CUDF_CUDA_TRY(cudaEventCreateWithFlags(&e_, cudaEventDisableTiming)); }
-  virtual ~cuda_event() { CUDF_ASSERT_CUDA_SUCCESS(cudaEventDestroy(e_)); }
```

Comment:
*issue (maybe)*: This means that _every_ usage of `cuda_event` in the code base will leak the underlying event. Is this really what we want? Perhaps we only want to leak the `thread_local` ones below.

That is, should we change `event_for_thread` to do (approximately):

```
thread_local std::vector<cuda_event *> thread_events(get_num_cuda_devices());
...
thread_events[device_id.value()] = new cuda_event();
```
?

Alternately, if the _only_ usage of `cuda_event` is in `event_for_thread`, we could just delete this class completely and implement `event_for_thread` as:

```
thread_local std::vector<cudaEvent_t> thread_events(...);
if (!thread_events[device_id.value()]) {
    CUDF_CUDA_TRY(cudaEventCreateWithFlags(&thread_events[device_id.value()], ...);
}
return thread_events[device_id.value()];
```

WDYT?

Reply:
`cuda_event` is only used as the RAII wrapper by `event_for_thread`, so deleting the user-defined destructor and allocating `cuda_event` on the heap happen to be equivalent. I agree that the leak makes the `cuda_event` wrapper entirely superfluous, so I will use the alternative approach you suggested to make the code cleaner. Thank you @wence- !

Reply:
Completed